File: leetcode/find-common-characters.cpp
#include <bits/stdc++.h>


using namespace std;
class Solution {
public:
vector<string> commonChars(vector<string>& words) {
    vector<string> fo;
    unordered_map<char, int> fd;
    unordered_map<char, int> d;
    for (char c : words[0]) {
        if (fd.find(c) == fd.end()) {
            fd[c] = 1;
        }
        else {
            fd[c]++;
        }
    }

    for (string word : words) {
        d.clear();
        for(char c:word) {
            if (d.find(c) == d.end()) {
                d[c] = 1;
            }
            else {d[c]++;}
        }
        for(const auto& pair:fd) {
            if (d.find(pair.first) == d.end()) {
                fd[pair.first] = 0;
            }
        }

        for(const auto& pair:fd){
            if(d[pair.first]<fd[pair.first]){
                fd[pair.first]=d[pair.first];
            }
        }
    }
    for(const auto& pair:fd){
        for(int i = 0; i<pair.second; i++){
            fo.push_back(string(1, pair.first));
        }
    }


return fo;


}};

File: leetcode/find-common-characters.py
class Solution:
    def commonChars(self, words: list[str]) -> list[str]:
        fo = []
        fd = {}

        for x in words[0]:
            if x in fd:
                fd[x] += 1
            else:
                fd[x] = 1

        for word in words:
            d = {}
            for char in word:
                if char in d:
                    d[char] += 1
                else:
                    d[char] = 1

            for key in fd:
                if key not in d:
                    fd[key] = 0

            for key in d:
                if key in fd:
                    if fd[key] > d[key]:
                        fd[key] = d[key]

        for x in fd:
            for z in range(fd[x]):
                fo.append(x)
        return fo
